1.28: fails to build due to: undefined reference to json_gen_* errors

Installed json-c is 0.18, using the release tarball https://github.com/containers/crun/releases/download/1.28/crun-1.28.tar.gz, configure also picks it up correctly:

```
[...]
checking for json-c >= 0.14... yes
[...]
```

But then runs into a build failure:

```
[...]
bin/sh ./libtool  --tag=CC   --mode=link x86_64-pc-linux-gnu-cc -I /var/tmp/paludis/build/app-virtualization-crun-1.28/work/crun-1.28/libocispec/src -I /var/tmp/paludis/build/app-virtualization-crun-1.28/work/crun-1.28/libocispec/src -I/usr/x86_64-pc-linux-gnu/include/json-c -D CRUN_LIBDIR="\"/usr/x86_64-pc-linux-gnu/lib/crun\""  -march=native -O2 -pipe -fPIC  -Wl,-O1 -Wl,--as-needed -fPIC -o crun src/crun-crun.o src/crun-run.o src/crun-delete.o src/crun-kill.o src/crun-pause.o src/crun-unpause.o src/crun-oci_features.o src/crun-spec.o src/crun-exec.o src/crun-list.o src/crun-create.o src/crun-start.o src/crun-state.o src/crun-update.o src/crun-ps.o src/crun-checkpoint.o src/crun-restore.o src/crun-mounts.o src/crun-run_create.o libcrun.la -lsystemd -lseccomp -lcap  -lm -ljson-c 
libtool: link: x86_64-pc-linux-gnu-cc -I /var/tmp/paludis/build/app-virtualization-crun-1.28/work/crun-1.28/libocispec/src -I /var/tmp/paludis/build/app-virtualization-crun-1.28/work/crun-1.28/libocispec/src -I/usr/x86_64-pc-linux-gnu/include/json-c -D CRUN_LIBDIR=\"/usr/x86_64-pc-linux-gnu/lib/crun\" -march=native -O2 -pipe -fPIC -Wl,-O1 -Wl,--as-needed -fPIC -o .libs/crun src/crun-crun.o src/crun-run.o src/crun-delete.o src/crun-kill.o src/crun-pause.o src/crun-unpause.o src/crun-oci_features.o src/crun-spec.o src/crun-exec.o src/crun-list.o src/crun-create.o src/crun-start.o src/crun-state.o src/crun-update.o src/crun-ps.o src/crun-checkpoint.o src/crun-restore.o src/crun-mounts.o src/crun-run_create.o  ./.libs/libcrun.so -lsystemd /usr/x86_64-pc-linux-gnu/lib/libseccomp.so -lcap -lm -ljson-c -Wl,-rpath -Wl,/usr/x86_64-pc-linux-gnu/lib
make[2]: Leaving directory '/var/tmp/paludis/build/app-virtualization-crun-1.28/work/crun-1.28'
make[1]: Leaving directory '/var/tmp/paludis/build/app-virtualization-crun-1.28/work/crun-1.28'
/usr/x86_64-pc-linux-gnu/x86_64-pc-linux-gnu/bin/ld.bfd: src/crun-oci_features.o: in function `add_string_to_json':
oci_features.c:(.text+0x45): undefined reference to `json_gen_string'
/usr/x86_64-pc-linux-gnu/x86_64-pc-linux-gnu/bin/ld.bfd: src/crun-oci_features.o: in function `add_bool_to_json':
oci_features.c:(.text+0x94): undefined reference to `json_gen_string'
/usr/x86_64-pc-linux-gnu/x86_64-pc-linux-gnu/bin/ld.bfd: src/crun-oci_features.o: in function `add_bool_str_to_json':
oci_features.c:(.text+0xe6): undefined reference to `json_gen_string'
/usr/x86_64-pc-linux-gnu/x86_64-pc-linux-gnu/bin/ld.bfd: src/crun-oci_features.o: in function `add_array_to_json':
oci_features.c:(.text+0x12f): undefined reference to `json_gen_string'
/usr/x86_64-pc-linux-gnu/x86_64-pc-linux-gnu/bin/ld.bfd: oci_features.c:(.text+0x137): undefined reference to `json_gen_array_open'
/usr/x86_64-pc-linux-gnu/x86_64-pc-linux-gnu/bin/ld.bfd: oci_features.c:(.text+0x166): undefined reference to `json_gen_string'
/usr/x86_64-pc-linux-gnu/x86_64-pc-linux-gnu/bin/ld.bfd: src/crun-oci_features.o: in function `crun_features_add_cgroup_info':
oci_features.c:(.text+0x1d9): undefined reference to `json_gen_string'
/usr/x86_64-pc-linux-gnu/x86_64-pc-linux-gnu/bin/ld.bfd: oci_features.c:(.text+0x1e1): undefined reference to `json_gen_map_open'
/usr/x86_64-pc-linux-gnu/x86_64-pc-linux-gnu/bin/ld.bfd: src/crun-oci_features.o: in function `crun_features_add_seccomp_info':
oci_features.c:(.text+0x259): undefined reference to `json_gen_string'
/usr/x86_64-pc-linux-gnu/x86_64-pc-linux-gnu/bin/ld.bfd: oci_features.c:(.text+0x261): undefined reference to `json_gen_map_open'
/usr/x86_64-pc-linux-gnu/x86_64-pc-linux-gnu/bin/ld.bfd: src/crun-oci_features.o: in function `crun_features_add_mempolicy_info':
oci_features.c:(.text+0x2d9): undefined reference to `json_gen_string'
/usr/x86_64-pc-linux-gnu/x86_64-pc-linux-gnu/bin/ld.bfd: oci_features.c:(.text+0x2e1): undefined reference to `json_gen_map_open'
/usr/x86_64-pc-linux-gnu/x86_64-pc-linux-gnu/bin/ld.bfd: src/crun-oci_features.o: in function `crun_features_add_apparmor_info':
oci_features.c:(.text+0x349): undefined reference to `json_gen_string'
/usr/x86_64-pc-linux-gnu/x86_64-pc-linux-gnu/bin/ld.bfd: oci_features.c:(.text+0x351): undefined reference to `json_gen_map_open'
/usr/x86_64-pc-linux-gnu/x86_64-pc-linux-gnu/bin/ld.bfd: src/crun-oci_features.o: in function `crun_features_add_selinux_info':
oci_features.c:(.text+0x399): undefined reference to `json_gen_string'
/usr/x86_64-pc-linux-gnu/x86_64-pc-linux-gnu/bin/ld.bfd: oci_features.c:(.text+0x3a1): undefined reference to `json_gen_map_open'
/usr/x86_64-pc-linux-gnu/x86_64-pc-linux-gnu/bin/ld.bfd: src/crun-oci_features.o: in function `crun_features_add_mount_ext_info':
oci_features.c:(.text+0x3e9): undefined reference to `json_gen_string'
/usr/x86_64-pc-linux-gnu/x86_64-pc-linux-gnu/bin/ld.bfd: oci_features.c:(.text+0x3f1): undefined reference to `json_gen_map_open'
/usr/x86_64-pc-linux-gnu/x86_64-pc-linux-gnu/bin/ld.bfd: oci_features.c:(.text+0x405): undefined reference to `json_gen_string'
/usr/x86_64-pc-linux-gnu/x86_64-pc-linux-gnu/bin/ld.bfd: oci_features.c:(.text+0x40d): undefined reference to `json_gen_map_open'
/usr/x86_64-pc-linux-gnu/x86_64-pc-linux-gnu/bin/ld.bfd: oci_features.c:(.text+0x428): undefined reference to `json_gen_map_close'
/usr/x86_64-pc-linux-gnu/x86_64-pc-linux-gnu/bin/ld.bfd: src/crun-oci_features.o: in function `crun_features_add_intel_rdt':
oci_features.c:(.text+0x459): undefined reference to `json_gen_string'
/usr/x86_64-pc-linux-gnu/x86_64-pc-linux-gnu/bin/ld.bfd: oci_features.c:(.text+0x461): undefined reference to `json_gen_map_open'
/usr/x86_64-pc-linux-gnu/x86_64-pc-linux-gnu/bin/ld.bfd: src/crun-oci_features.o: in function `crun_features_add_net_devices':
oci_features.c:(.text+0x4a9): undefined reference to `json_gen_string'
/usr/x86_64-pc-linux-gnu/x86_64-pc-linux-gnu/bin/ld.bfd: oci_features.c:(.text+0x4b1): undefined reference to `json_gen_map_open'
/usr/x86_64-pc-linux-gnu/x86_64-pc-linux-gnu/bin/ld.bfd: src/crun-oci_features.o: in function `crun_features_add_linux_info':
oci_features.c:(.text+0x4f9): undefined reference to `json_gen_string'
/usr/x86_64-pc-linux-gnu/x86_64-pc-linux-gnu/bin/ld.bfd: oci_features.c:(.text+0x501): undefined reference to `json_gen_map_open'
/usr/x86_64-pc-linux-gnu/x86_64-pc-linux-gnu/bin/ld.bfd: src/crun-oci_features.o: in function `crun_features_add_annotations_info':
oci_features.c:(.text+0x5a9): undefined reference to `json_gen_string'
/usr/x86_64-pc-linux-gnu/x86_64-pc-linux-gnu/bin/ld.bfd: oci_features.c:(.text+0x5b1): undefined reference to `json_gen_map_open'
/usr/x86_64-pc-linux-gnu/x86_64-pc-linux-gnu/bin/ld.bfd: src/crun-oci_features.o: in function `crun_command_features':
oci_features.c:(.text+0x6fc): undefined reference to `json_gen_init'
/usr/x86_64-pc-linux-gnu/x86_64-pc-linux-gnu/bin/ld.bfd: oci_features.c:(.text+0x718): undefined reference to `json_gen_config'
/usr/x86_64-pc-linux-gnu/x86_64-pc-linux-gnu/bin/ld.bfd: oci_features.c:(.text+0x722): undefined reference to `json_gen_map_open'
/usr/x86_64-pc-linux-gnu/x86_64-pc-linux-gnu/bin/ld.bfd: oci_features.c:(.text+0x7ba): undefined reference to `json_gen_map_close'
/usr/x86_64-pc-linux-gnu/x86_64-pc-linux-gnu/bin/ld.bfd: oci_features.c:(.text+0x7d5): undefined reference to `json_gen_get_buf'
/usr/x86_64-pc-linux-gnu/x86_64-pc-linux-gnu/bin/ld.bfd: oci_features.c:(.text+0x7f0): undefined reference to `json_gen_free'
/usr/x86_64-pc-linux-gnu/x86_64-pc-linux-gnu/bin/ld.bfd: src/crun-oci_features.o: in function `add_string_to_json':
oci_features.c:(.text+0x61): undefined reference to `json_gen_string'
/usr/x86_64-pc-linux-gnu/x86_64-pc-linux-gnu/bin/ld.bfd: src/crun-oci_features.o: in function `add_bool_to_json':
oci_features.c:(.text+0xa4): undefined reference to `json_gen_bool'
/usr/x86_64-pc-linux-gnu/x86_64-pc-linux-gnu/bin/ld.bfd: src/crun-oci_features.o: in function `add_bool_str_to_json':
oci_features.c:(.text+0x102): undefined reference to `json_gen_string'
/usr/x86_64-pc-linux-gnu/x86_64-pc-linux-gnu/bin/ld.bfd: src/crun-oci_features.o: in function `add_array_to_json':
oci_features.c:(.text+0x17b): undefined reference to `json_gen_array_close'
/usr/x86_64-pc-linux-gnu/x86_64-pc-linux-gnu/bin/ld.bfd: src/crun-oci_features.o: in function `crun_features_add_cgroup_info':
oci_features.c:(.text+0x23b): undefined reference to `json_gen_map_close'
/usr/x86_64-pc-linux-gnu/x86_64-pc-linux-gnu/bin/ld.bfd: src/crun-oci_features.o: in function `crun_features_add_seccomp_info':
oci_features.c:(.text+0x2b2): undefined reference to `json_gen_map_close'
/usr/x86_64-pc-linux-gnu/x86_64-pc-linux-gnu/bin/ld.bfd: src/crun-oci_features.o: in function `crun_features_add_mempolicy_info':
oci_features.c:(.text+0x31f): undefined reference to `json_gen_map_close'
/usr/x86_64-pc-linux-gnu/x86_64-pc-linux-gnu/bin/ld.bfd: src/crun-oci_features.o: in function `crun_features_add_apparmor_info':
oci_features.c:(.text+0x372): undefined reference to `json_gen_map_close'
/usr/x86_64-pc-linux-gnu/x86_64-pc-linux-gnu/bin/ld.bfd: src/crun-oci_features.o: in function `crun_features_add_selinux_info':
oci_features.c:(.text+0x3c2): undefined reference to `json_gen_map_close'
/usr/x86_64-pc-linux-gnu/x86_64-pc-linux-gnu/bin/ld.bfd: src/crun-oci_features.o:oci_features.c:(.text+0x436): more undefined references to `json_gen_map_close' follow
collect2: error: ld returned 1 exit status
make[2]: *** [Makefile:1859: crun] Error 1
make[1]: *** [Makefile:3182: all-recursive] Error 1
make: *** [Makefile:1324: all] Error 2
```

complete build log: [crun-1.28_build.log](https://github.com/user-attachments/files/28314478/crun-1.28_build.log)

--- a/.github/workflows/test.yaml
+++ b/.github/workflows/test.yaml
@@ -53,7 +53,7 @@ jobs:
 
             make -j $(nproc) clean
 
-            if ./configure --enable-embedded-blake3 CFLAGS='-Wall -Werror --enable-shared'; then
+            if ./configure --enable-embedded-blake3 --enable-shared CFLAGS='-Wall -Werror'; then
                         make -j $(nproc) -C libocispec libocispec.la
                         make git-version.h
                         make -j $(nproc) libcrun.la
@@ -68,6 +68,7 @@ jobs:
         include:
           - test: disable-systemd
           - test: check
+          - test: enable-shared
           - test: embedded-blake3
           - test: system-blake3
           - test: podman
@@ -166,6 +167,10 @@ jobs:
               clang-check)
                   sudo docker build -t crun-clang-check tests/clang-check
                   sudo docker run --privileged --rm -w /crun -v ${PWD}:/crun crun-clang-check
+              ;;
+              enable-shared)
+                  ./configure --enable-embedded-blake3 --enable-shared
+                  make -j $(nproc)
               ;;
               embedded-blake3)
                   ./configure --enable-embedded-blake3

--- a/libcrun.lds
+++ b/libcrun.lds
@@ -6,5 +6,6 @@
         runtime_spec_*;
         free_runtime_spec_*;
         make_runtime_spec_*;
+        json_gen_*;
   local: *;
 };

